client/post-editor/media-modal/index.jsx

```diff
@@ -174,6 +180,11 @@ export const EditorMediaModal = React.createClass( {
 
 		MediaActions.delete( site.ID, toDelete );
 		analytics.mc.bumpStat( 'editor_media_actions', 'delete_media' );
+		const mediaIds = map( toDelete, 'ID' );
+		this.props.deleteMedia( site.ID, mediaIds );
+		if ( includes( mediaIds, siteIconId ) ) {
+			this.props.resetSiteIcon( site.ID );
+		}
```

Comment:
@aduth middleware is probably a reasonable solution for this, but it _would_ seem a bit odd with the `deleteMedia` action doing the deletion and leaving the site icon in middleware. maybe not. maybe it would actually be a quick and easy function to add to the wpcom data layer.

in essence, we should ask if deleting the media item currently set as the site icon _always_ means that the icon should be reset. if this is the case, I would expect to have some `siteIcon` reducer somewhere in the state tree listening for `mediaDelete` actions and updating in Calypso-first by resetting it. in other words, `deleteMedia` would _not_ be the genesis for resetting the site icon. this is by design in Redux. the `siteIcon` state/reducer is responsible for its own behavior and if the proposition holds, it should be listening for all relevant actions which affect it.

the middleware comes in as the _data layer_. we act as if Calypso is the raw source of truth. the site icon reducer resets its value when it see `MEDIA_DELETE` with its own media id. the middleware layer for the site icon (or site setting) endpoint also listens for this same action and will trigger an API call whenever one flies by. if the API call fails, it may very well reset the site icon to its previous id, or in this case, if the media is gone, the failure mode will do something else - set some default for example.

does this help?